### PR TITLE
Kiota with dependency injection dotnet

### PR DESCRIPTION
**Pull request by svrooij**
Sample code for: https://github.com/MicrosoftDocs/openapi-docs/pull/89

Building and updating it, is not implemented. Want me to update the workflows as well?

@baywet here you go

### CODE REVIEW

**Review by mnMomma — APPROVED** [review]

**Review by baywet — CHANGES_REQUESTED** [review]
Thank you for the contribution!
Can I ask that you please:
- add a configuration in dependabot
- add a github workflow (build only)

**Comment by svrooij** [comment]
> Thank you for the contribution! Can I ask that you please:
> 
> * add a configuration in dependabot
> * add a github workflow (build only)

Sure, here you go.

**Comment by svrooij** [comment]
@baywet are you able to have a look at my latest changes? If this is accepted I can also change the other pr to use the code in this sample

File: .github/workflows/get-started-dotnet.yml [hunk 1]
```diff
@@ -55,6 +57,24 @@ jobs:
       run: dotnet build --no-restore
       working-directory: get-started/azure-auth/cli/src/
 
+  build-dotnet-dependency-injection:
+
+    runs-on: ubuntu-latest
```
Comment (baywet) [inline .github/workflows/get-started-dotnet.yml:62]:
```suggestion
    runs-on: ubuntu-latest
    needs: []
```

(and replicate to other jobs please so they run in parallels and not sequentially)

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for making the changes, and for your patience (I was out)
One last thing and we should be good to go.

File: .github/workflows/get-started-dotnet.yml (hunk 1 above)
Comment (svrooij) [reply .github/workflows/get-started-dotnet.yml:64]: I changed all the jobs to have the same structure (where the actions will run in parallel).

This is commit 7, you might want to squash this PR

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

File: .github/workflows/get-started-dotnet.yml [hunk 2]
```diff
@@ -55,9 +59,29 @@ jobs:
...
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 8.x
+
+    - name: Restore dependencies dotnet-dependency-injection
+      run: dotnet restore
```
Comment (baywet) [inline .github/workflows/get-started-dotnet.yml:75]:
```suggestion
      run: dotnet restore ./dotnet-dependency-injection.sln
```

File: .github/workflows/get-started-dotnet.yml [hunk 3]
```diff
@@ -55,9 +59,29 @@ jobs:
...
+      working-directory: get-started/dotnet-dependency-injection/
+    - name: Build dotnet-dependency-injection
+      run: dotnet build --no-restore
```
Comment (baywet) [inline .github/workflows/get-started-dotnet.yml:78]:
```suggestion
      run: dotnet build ./dotnet-dependency-injection.sln --no-restore
```

**Review by baywet — APPROVED** [review]

**Comment by baywet** [comment]
@svrooij the project doesn't seem to be building anymore for some reason, can you have a look please?

**Review by baywet — APPROVED** [review]

**Review by baywet — APPROVED** [review]

**Comment by baywet** [comment]
@svrooij just making sure we don't loose momentum here :)

**Comment by svrooij** [comment]
@baywet i literally have no clue why it doesn't work.... It builds in Ubuntu on wsl and it builds on windows

**Comment by baywet** [comment]
@svrooij it wasn't building on my machine either. Turns out files were missing for some reason. Generating and pushing files fixed it. Thank you for your patience.

**Pull request merged**